### PR TITLE
Add missing cast which causes pointer to address not to be loaded

### PR DESCRIPTION
**Pull request by seanyoung**
Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

**Comment by seanyoung** [comment]
Make solidity in issue #678 compile

File: src/sema/expression.rs [hunk 1]
```diff
@@ -5995,21 +5995,30 @@ fn method_call_pos_args(
                 ResolveTo::Type(&Type::Value),
             )?;
 
+            let address = cast(
+                &var_expr.loc(),
+                var_expr,
```
Comment (LucasSte) [inline src/sema/expression.rs:6000]: Is there any case in which `loc` is not the same `loc` saved in an expression for the `cast` function? It seems redundant to have these two arguments as the former is an attribute of the latter.

Comment (seanyoung) [reply src/sema/expression.rs:6000]: That is a good point, I had not considered that. 

I think you are totally right and this is remnant of the way the code used to be structured. 

Let me see if I remove the argument.

Comment (seanyoung) [reply src/sema/expression.rs:6000]: I started this work and now I found a case where we use a different location for the first argument and the expression. You can see that here.

image

**Review by LucasSte — APPROVED** [review]

**Pull request merged**